Sources/RESP3/RESP3Token.swift

```diff
@@ -0,0 +1,528 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the RediStack open source project
+//
+// Copyright (c) 2023 RediStack project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of RediStack project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+public struct RESP3Token: Hashable, Sendable {
+    public struct Array: Sequence, Sendable, Hashable {
+        public typealias Element = RESP3Token
+
+        public let count: Int
+        let buffer: ByteBuffer
+
+        public func makeIterator() -> Iterator {
+            Iterator(buffer: self.buffer)
+        }
+
+        public struct Iterator: IteratorProtocol {
+            public typealias Element = RESP3Token
+
+            private var buffer: ByteBuffer
+
+            fileprivate init(buffer: ByteBuffer) {
+                self.buffer = buffer
+            }
+
+            public mutating func next() -> RESP3Token? {
+                return try! RESP3Token(consuming: &self.buffer)
```

Comment:
Did you unit test this iterator? I believe this to be bugged, as we need to move to the first element before consuming. Otherwise `next()` would return the whole array as the only element I believe

Reply:
yes, we do have tests for this. the array only gets the elements slice of the array.

Reply:
I guess I overlooked it then